tests/functional/labels.js

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern',
+  'intern!object',
+  'intern/chai!assert',
+  'require'
+], function (intern, registerSuite, assert, require) {
+  'use strict';
+
+  var url = function (num) {
+    return intern.config.siteRoot + '/issues/' + num;
+  };
+
+  registerSuite({
+    name: 'labels',
+
+    setup: function () {
+      // We should be logged before starting the labels test.
+      return this.remote
+        .setFindTimeout(intern.config.wc.pageLoadTimeout)
+        .get(require.toUrl(url(100)))
+        .findByCssSelector('.js-login-link').click()
```

Comment:
Yes you are right. The test has to be better. 
In the setup I should probably pseudo-code

``` js
loggedin: function checkLogStatus();

if (loggedin) { 
  pass;
} elsif { 
  if (sessionCookies) {
    loginLinkClick();
  } else {
    logMePlease(login, passwd);
  }
};
// let's make sure we are logged before proceeding
if !(loggedin) {
  quitWithErrorMsg('Something is rotten');
}
```